teuthology/scrape.py

```diff
@@ -410,7 +410,10 @@ def _get_service_types(self, job):
         result = defaultdict(list)
         # Lines like:
         # 2014-08-22T20:07:18.668 ERROR:tasks.ceph:saw valgrind issue   <kind>Leak_DefinitelyLost</kind> in /var/log/ceph/valgrind/osd.3.log.gz
-        for line in grep(os.path.join(job.path, "teuthology.log"), "</kind> in "):
+        valgrind_err_line = grep(os.path.join(job.path, "teuthology.log"), "</kind> in ")
+        # removes blank space, empty string and None
+        valgrind_err_line = [line for line in valgrind_err_line if line and line.strip()]
```

Comment:
teuthology logs are usually huge, and _grep_ function is already returns list of lines, using of list comprehension is quite expensive here, maybe it is better to use a generator expression?
but I've got more general question, is there any sense to call grep in `Call out to native grep rather than feeding massive log files through python line by line`, are python regex are that slow?

Reply:
In my benchmarks, using a list comprehension is ~2x as fast as grep for a statement like this.

Reply:
My points were:
1) use generator expression instead of list comprehension here.
2) remove scrape.grep methods and replace with pure python equivalent.

Reply:
I see, that's more clear. I don't disagree, but I think replacing `grep()` can be considered a separate issue from this PR.

Reply:
> I see, that's more clear. I don't disagree, but I think replacing `grep()` can be considered a separate issue from this PR.

There are just two places where grep is used, this place and another one with similar logic. At least with this PR in this place just drop grep usage.

Reply:
or, do you want to merge this PR as is?